php/meterpreter/ext_server_stdapi.php

```diff
@@ -429,25 +429,40 @@ function add_stat_buf($path) {
     if ($st) {
         $st_buf = "";
         $st_buf .= pack("V", $st['dev']);
-        $st_buf .= pack("v", $st['ino']);
-        $st_buf .= pack("v", $st['mode']);
-        $st_buf .= pack("v", 0);
-        $st_buf .= pack("v", $st['nlink']);
-        $st_buf .= pack("v", $st['uid']);
-        $st_buf .= pack("v", $st['gid']);
+        $st_buf .= pack("V", $st['mode']);
+        $st_buf .= pack("V", $st['nlink']);
+        $st_buf .= pack("V", $st['uid']);
+        $st_buf .= pack("V", $st['gid']);
         $st_buf .= pack("V", $st['rdev']);
-        $st_buf .= pack("V", $st['size']);
-        $st_buf .= pack("V", $st['ctime']);
-        $st_buf .= pack("V", $st['atime']);
-        $st_buf .= pack("V", $st['mtime']);
+
+        $st_buf .= pack_p($st['ino']);
+        $st_buf .= pack_p($st['size']);
+        $st_buf .= pack_p($st['atime']);
+        $st_buf .= pack_p($st['mtime']);
+        $st_buf .= pack_p($st['ctime']);
+        
         $st_buf .= pack("V", $st['blksize']);
         $st_buf .= pack("V", $st['blocks']);
-        return create_tlv(TLV_TYPE_STAT_BUF32, $st_buf);
+       
+        return create_tlv(TLV_TYPE_STAT_BUF, $st_buf);
     }
     return false;
 }
 }
 
+if(!function_exists('pack_p')) {
+# Implements pack('P', $value) - but backwards compatible to PHP4.x
+# https://www.php.net/manual/en/function.pack.php
+# Directive:
+#   P   unsigned long long (always 64 bit, little endian byte order)
+function pack_p($value) {
+    $first_half = pack('V', $value & 0xffffffff);
+    $second_half = pack('V', ($value >> 32) & 0xffffffff);
```

Comment:
This seems more like your dropping the lower 32 bits by bit shifting the lower 32 bits off the end of the line, and then doing a weird operation of a logical AND on the result to retain only the lower 32 bits of whatever remains.

If anything this seems more like the operation to get the top 32 bits of the number?